crates/accelerate/src/convert_2q_block_matrix.rs

```diff
@@ -32,32 +38,63 @@ pub fn blocks_to_matrix(
     py: Python,
     op_list: Vec<(PyReadonlyArray2<Complex64>, SmallVec<[u8; 2]>)>,
 ) -> PyResult<Py<PyArray2<Complex64>>> {
-    let identity = aview2(&ONE_QUBIT_IDENTITY);
-    let input_matrix = op_list[0].0.as_array();
-    let mut matrix: Array2<Complex64> = match op_list[0].1.as_slice() {
-        [0] => kron(&identity, &input_matrix),
-        [1] => kron(&input_matrix, &identity),
-        [0, 1] => input_matrix.to_owned(),
-        [1, 0] => change_basis(input_matrix),
-        [] => Array2::eye(4),
+    let identity = aview2(&ONE_QUBIT_IDENTITY).into_faer_complex();
+    let input_matrix = op_list[0].0.as_array().into_faer_complex();
+    let mut matrix: Mat<c64> = Mat::<c64>::zeros(4, 4);
+
+    match op_list[0].1.as_slice() {
+        [0] => kron(matrix.as_mut(), identity.as_ref(), input_matrix.as_ref()),
+        [1] => kron(matrix.as_mut(), input_matrix.as_ref(), identity.as_ref()),
+        [0, 1] => matrix = input_matrix.to_owned(),
+        [1, 0] => matrix = change_basis_faer(input_matrix.to_owned()),
+        [] => matrix = Mat::<c64>::identity(4, 4),
```

Comment:
I assume you did `zeros()` here because using the `kron()` function is allocation free. But if you changed this to the form:
```suggestion
    let mut matrix: Mat<c64> = match op_list[0].1.as_slice() {
        [0] => identity.kron(input_matrix.as_ref()),
        [1] => input_matrix.kron(identity.as_ref()),
        [0, 1] => input_matrix.to_owned(),
        [1, 0] => change_basis_faer(input_matrix.to_owned()),
        [] => matrix = Mat::<c64>::identity(4, 4),
```
by using the kron method on a matrix object (https://docs.rs/faer/latest/faer/mat/struct.Mat.html#method.kron) it would be a bit more idomatic and avoid an allocation in the 2q case.

Reply:
For the `kron` method, similar to `matmul`, I ran the same tests as in the issue, and, curiously, using the version without allocation has a negative impact as well.

|       |**% change avg (no-allocation -> allocation)** | no-allocation  (min) | no-allocation  (max) | **no-allocation  (avg)** | allocation (min) | allocation (max) | **allocation (avg)** |
|:-----:|:---------------:|:---------------:|:-------------:|:-------------------:|:------------:|:------------:|:----------------:|
|  2x2  |    **47.92%**        |            848ns     |     1.94µs     |    **1.08µs**   |    427ns   |    815ns   |  **562.5ns**  |
|  4x4  |    **63.16%**       |        2.62µs     |     5.82µs     |    **3.04µs**   |   945ns  |   1.54µs   |  **1.12µs**  |